modules/tinymce/src/core/test/ts/browser/html/SerializerTest.ts

```diff
@@ -28,14 +28,13 @@ describe('browser.tinymce.core.html.SerializerTest', () => {
     );
   });
 
-  // TODO: TINY-4627/TINY-8383
-  it.skip('Serialize with validate: true, when parsing with validate:false bug', () => {
+  it('Serialize with validate: true, when parsing with validate:false bug', () => {
     const schema = Schema({ valid_elements: 'b' });
     const serializer = HtmlSerializer({}, schema);
 
     assert.equal(
       serializer.serialize(DomParser({ validate: false }, schema).parse('<b a="1" b="2">a</b><i a="1" b="2">b</i>')),
-      '<b a="1" b="2">a</b><i a="1" b="2">b</i>'
+      '<b b="2" a="1">a</b><i b="2" a="1">b</i>'
```

Comment:
Why is this in reverse order now? Should we change the HtmlWriter to sort them even when validate is false to combat this. Could other tests be failing now since it no longer producing the same output as before?

Reply:
This was something discussed in stand ups sorry, but we said we weren't concerned about attribute order for things like this as it's not part of the HTML spec. It's also not related to `validate: false` and that's not set for the serializer. So if the order is defined by the schema then it will order them appropriately, but if not it'll just return them however the native parser orders them (or however DOMPurify reorganizes them).

Reply:
I should add I'm not exactly sure why it's in the reverse order, but this had come up in Neil's previous PRs and he had attributed it to the above so I assumed the same. If you're concerned I can look into it more, but it wasn't something I thought we needed to worry about.

Reply:
Ok. I just get worried when I see tests being changed and not knowing the context of that change.

Reply:
So I checked why this has reversed and it's because of how DOMPurify does it's sanitization of attributes here: https://github.com/cure53/DOMPurify/blob/84f1d45/src/purify.js#L1146. Basically it'll walk backwards and remove than re-add valid attributes, which in turn causes the attribute order to be reversed.

I can't see any good reason why they forcibly remove it and add it back, but the reverse order makes sense to account for the fact that removing an attribute would cause iteration issues if going forwards. We might be able to submit a change to only remove if required which would prevent it reordering if we're really concerned.